core/src/test/java/de/danoeh/antennapod/core/util/playback/TimelineTest.java

```diff
@@ -15,29 +14,40 @@
 import de.danoeh.antennapod.core.feed.Chapter;
 import de.danoeh.antennapod.core.feed.FeedItem;
 import de.danoeh.antennapod.core.feed.FeedMedia;
-import de.danoeh.antennapod.core.util.playback.Playable;
-import de.danoeh.antennapod.core.util.playback.Timeline;
+import de.danoeh.antennapod.core.storage.DBReader;
+
 import org.junit.Before;
+import org.junit.BeforeClass;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.robolectric.RobolectricTestRunner;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 
 /**
- * Test class for timeline.
+ * Test class for {@link Timeline}.
  */
-@SmallTest
+@RunWith(RobolectricTestRunner.class)
 public class TimelineTest {
 
     private Context context;
 
+    @BeforeClass
+    public static void beforeClass() {
+        //noinspection ResultOfMethodCallIgnored
+        Mockito.mockStatic(DBReader.class);
```

Comment:
Do you think this could somehow interfere with the other tests, causing the NPEs?

Reply:
Great hint, that would explain why I couldn't reproduce the NPEs in issue #4816.
I updated the test code.

Reply:
Nice! Can this also be done in the before/after methods to have a bit less code duplication? I'm not too experienced with Mockito, so maybe this does not work cleanly. If we decouple the classes later, the mock might not be needed anymore, so we would have to revert everything again. Just removing the before/after calls would be a bit less invasive.

Reply:
Sure, I refactored the code and it works fine, too. Actually, that makes the code more clear.